test_unhandled_sqlite_misuse: AssertionError: False is not true : this test failing means either the sqlite race conditions have been fixed in cpython or the test max_attempts needs to be increased

https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726

```
sqlite3 2.6.0/python 3.7.13 (default, May  4 2022, 08:45:32) did not raise SQLITE_MISUSE within 80000 attempts of the race condition
```

```
478
======================================================================
[479](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:480)
FAIL: test_unhandled_sqlite_misuse (tests.unit.wallet.test_database.TestSQLiteRace)
[480](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:481)
----------------------------------------------------------------------
[481](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:482)
Traceback (most recent call last):
[482](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:483)
  File "/home/runner/work/lbry-sdk/lbry-sdk/lbry/testcase.py", line 145, in run
[483](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:484)
    self.loop.run_until_complete(maybe_coroutine)
[484](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:485)
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
[485](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:486)
    return future.result()
[486](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:487)
  File "/home/runner/work/lbry-sdk/lbry-sdk/tests/unit/wallet/test_database.py", line 535, in test_unhandled_sqlite_misuse
[487](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:488)
    self.assertTrue(False, 'this test failing means either the sqlite race conditions '
[488](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:489)
AssertionError: False is not true : this test failing means either the sqlite race conditions have been fixed in cpython or the test max_attempts needs to be increased
[489](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:490)

[490](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:491)
----------------------------------------------------------------------
[491](https://github.com/moodyjon/lbry-sdk/actions/runs/3091793824/jobs/5002309726#step:9:492)
Ran 348 tests in 214.348s
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -160,7 +160,7 @@ jobs:
           path: ${{ steps.pip-cache.outputs.dir }}
           key: ${{ runner.os }}-pip-${{ hashFiles('setup.py') }}
           restore-keys: ${{ runner.os }}-pip-
-      - run: pip install pyinstaller==4.4
+      - run: pip install pyinstaller==4.6
       - run: pip install -e .
       - if: startsWith(github.ref, 'refs/tags/v')
         run: python docker/set_build.py

--- a/lbry/wallet/orchstr8/node.py
+++ b/lbry/wallet/orchstr8/node.py
@@ -214,6 +214,7 @@ class SPVNode:
         self.port = 50001 + node_number  # avoid conflict with default daemon
         self.udp_port = self.port
         self.elastic_notifier_port = 19080 + node_number
+        self.elastic_services = f'localhost:9200/localhost:{self.elastic_notifier_port}'
         self.session_timeout = 600
         self.stopped = True
         self.index_name = uuid4().hex
@@ -235,7 +236,7 @@ class SPVNode:
                 'host': self.hostname,
                 'tcp_port': self.port,
                 'udp_port': self.udp_port,
-                'elastic_notifier_port': self.elastic_notifier_port,
+                'elastic_services': self.elastic_services,
                 'session_timeout': self.session_timeout,
                 'max_query_workers': 0,
                 'es_index_prefix': self.index_name,

--- a/lbry/wallet/wallet.py
+++ b/lbry/wallet/wallet.py
@@ -182,6 +182,8 @@ class Wallet:
                 raise InvalidPasswordError()
             if "unknown compression method" in e.args[0].lower():
                 raise InvalidPasswordError()
+            if "invalid window size" in e.args[0].lower():
+                raise InvalidPasswordError()
             raise
         return json.loads(decompressed)
 

--- a/setup.py
+++ b/setup.py
@@ -57,7 +57,7 @@ setup(
             'jsonschema==4.4.0',
         ],
         'hub': [
-            'hub@git+https://github.com/lbryio/hub.git@dcd4d7a7a8de3ab41d9fb859b910c3b962eaa2fe'
+            'hub@git+https://github.com/lbryio/hub.git@929448d64bcbe6c5e476757ec78456beaa85e56a'
         ]
     },
     classifiers=[

--- a/setup.py
+++ b/setup.py
@@ -57,7 +57,7 @@ setup(
             'jsonschema==4.4.0',
         ],
         'hub': [
-            'hub@git+https://github.com/lbryio/hub.git@9b178222296aecc7699cf82141c7a48fe866f22a'
+            'hub@git+https://github.com/lbryio/hub.git@dcd4d7a7a8de3ab41d9fb859b910c3b962eaa2fe'
         ]
     },
     classifiers=[

--- a/setup.py
+++ b/setup.py
@@ -57,7 +57,7 @@ setup(
             'jsonschema==4.4.0',
         ],
         'hub': [
-            'hub@git+https://github.com/lbryio/hub.git@024aceda53fe6d1ab8d519b73584437c25de6975'
+            'hub@git+https://github.com/lbryio/hub.git@9b178222296aecc7699cf82141c7a48fe866f22a'
         ]
     },
     classifiers=[

--- a/tests/integration/takeovers/test_resolve_command.py
+++ b/tests/integration/takeovers/test_resolve_command.py
@@ -1508,27 +1508,27 @@ class ResolveClaimTakeovers(BaseResolveTestCase):
         COIN = int(1E8)
 
         self.assertEqual(self.conductor.spv_node.writer.height, 207)
-        self.conductor.spv_node.writer.db.prefix_db.trending_notification.stage_put(
+        self.conductor.spv_node.writer.db.prefix_db.trending_notification.stash_put(
             (208, bytes.fromhex(claim_id1)), (0, 10 * COIN)
         )
         await self.generate(1)
         self.assertEqual(self.conductor.spv_node.writer.height, 208)
 
         self.assertEqual(1.7090807854206793, await get_trending_score(claim_id1))
-        self.conductor.spv_node.writer.db.prefix_db.trending_notification.stage_put(
+        self.conductor.spv_node.writer.db.prefix_db.trending_notification.stash_put(
             (209, bytes.fromhex(claim_id1)), (10 * COIN, 100 * COIN)
         )
         await self.generate(1)
         self.assertEqual(self.conductor.spv_node.writer.height, 209)
         self.assertEqual(2.2437974397778886, await get_trending_score(claim_id1))
-        self.conductor.spv_node.writer.db.prefix_db.trending_notification.stage_put(
+        self.conductor.spv_node.writer.db.prefix_db.trending_notification.stash_put(
             (309, bytes.fromhex(claim_id1)), (100 * COIN, 1000000 * COIN)
         )
         await self.generate(100)
         self.assertEqual(self.conductor.spv_node.writer.height, 309)
         self.assertEqual(5.157053472135866, await get_trending_score(claim_id1))
 
-        self.conductor.spv_node.writer.db.prefix_db.trending_notification.stage_put(
+        self.conductor.spv_node.writer.db.prefix_db.trending_notification.stash_put(
             (409, bytes.fromhex(claim_id1)), (1000000 * COIN, 1 * COIN)
         )
 

--- a/tests/integration/transactions/test_transaction_commands.py
+++ b/tests/integration/transactions/test_transaction_commands.py
@@ -2,7 +2,7 @@ import asyncio
 import unittest
 
 from lbry.testcase import CommandTestCase
-
+from lbry.wallet import Transaction
 
 class TransactionCommandsTestCase(CommandTestCase):
 
@@ -29,17 +29,42 @@ class TransactionCommandsTestCase(CommandTestCase):
         # someone's tx
         change_address = await self.blockchain.get_raw_change_address()
         sendtxid = await self.blockchain.send_to_address(change_address, 10)
-        await asyncio.sleep(0.2)
-        tx = await self.daemon.jsonrpc_transaction_show(sendtxid)
-        self.assertEqual(tx.id, sendtxid)
-        self.assertEqual(tx.height, -1)
+        # After a few tries, Hub should have the transaction (in mempool).
+        for i in range(5):
+            tx = await self.daemon.jsonrpc_transaction_show(sendtxid)
+            # Retry if Hub is not aware of the transaction.
+            if isinstance(tx, dict):
+                # Fields: 'success', 'code', 'message'
+                self.assertFalse(tx['success'], tx)
+                self.assertEqual(tx['code'], 404, tx)
+                self.assertEqual(tx['message'], "transaction not found", tx)
+                await asyncio.sleep(0.1)
+                continue
+            break
+        # verify transaction show (in mempool)
+        self.assertTrue(isinstance(tx, Transaction), str(tx))
+        # Fields: 'txid', 'raw', 'height', 'position', 'is_verified', and more.
+        self.assertEqual(tx.id, sendtxid, vars(tx))
+        self.assertEqual(tx.height, -1, vars(tx))
+        self.assertEqual(tx.is_verified, False, vars(tx))
+
+        # transaction is confirmed and leaves mempool
         await self.generate(1)
+
+        # verify transaction show
         tx = await self.daemon.jsonrpc_transaction_show(sendtxid)
-        self.assertEqual(tx.height, self.ledger.headers.height)
+        self.assertTrue(isinstance(tx, Transaction), str(tx))
+        self.assertEqual(tx.id, sendtxid, vars(tx))
+        self.assertEqual(tx.height, self.ledger.headers.height, vars(tx))
+        self.assertEqual(tx.is_verified, True, vars(tx))
 
         # inexistent
         result = await self.daemon.jsonrpc_transaction_show('0'*64)
-        self.assertFalse(result['success'])
+        self.assertTrue(isinstance(result, dict), result)
+        # Fields: 'success', 'code', 'message'
+        self.assertFalse(result['success'], result)
+        self.assertEqual(result['code'], 404, result)
+        self.assertEqual(result['message'], "transaction not found", result)
 
     async def test_utxo_release(self):
         await self.send_to_address_and_wait(

--- a/tests/unit/wallet/test_database.py
+++ b/tests/unit/wallet/test_database.py
@@ -470,7 +470,7 @@ class TestUpgrade(AsyncioTestCase):
 
 
 class TestSQLiteRace(AsyncioTestCase):
-    max_misuse_attempts = 80000
+    max_misuse_attempts = 120000
 
     def setup_db(self):
         self.db = sqlite3.connect(":memory:", isolation_level=None)

